### PR TITLE
Update service configuration file fix

### PR DESCRIPTION
**Pull request by shihchang**
Update service configuration file, parse out the transport from OsRegistry and set it in the update config

### CODE REVIEW

File: pkg/cluster/ignition/library.go [hunk 1]
```diff
@@ -251,12 +251,19 @@ func clusterCommon(cc *clusterCommonConfig, action string) (*igntypes.Config, er
 		Enabled: util.BoolPtr(true),
 	}
 
-	// Update service configuration file
+	// Update service configuration file, parse out the transport from OsRegistry and set it in the update config
+	transport := DefaultTransport
+	osRegistry := cc.OsRegistry
+	fields := strings.Split(cc.OsRegistry, ":")
```
Comment (prasad-shirodkar) [inline pkg/cluster/ignition/library.go:257]: Please use below snippet to detect the image transport mechanism:
```
import "github.com/containers/image/v5/transports/alltransports"
.....
.....
imageTransport := alltransports.TransportFromImageName(cc.BootVolumeContainerImage)
if imageTransport != nil {
   ......
   ......
}
```
We have used similar transport detection logic here https://github.com/oracle-cne/ocne/blob/f4cbeb3099e374cc432e919c2dea09527db6effb/cmd/cluster/start/start.go#L86

Comment (dkrasins) [reply pkg/cluster/ignition/library.go:257]: Does this function work correctly for ostree references?

Comment (shihchang) [reply pkg/cluster/ignition/library.go]: github.com/containers/image/v5/transports/alltransports does NOT recognize ostree transports:
Invalid image name "ostree-unverified-registry:container-registry.oracle.com/olcne/ock-ostree", unknown transport "ostree-unverified-registry"
Changed to ParseOstreeReference(https://github.com/oracle-cne/ocne/blob/main/pkg/image/registry.go#L51) as @dkrasins suggested

**Comment by dkrasins** [comment]
This PR has been superseded by https://github.com/oracle-cne/ocne/pull/19

**Pull request closed without merging**